Source/Core/DolphinQt/Config/Mapping/MappingNumeric.h

```diff
@@ -4,23 +4,34 @@
 
 #pragma once
 
-#include <QSpinBox>
+#include <QCheckBox>
+#include <QDoubleSpinBox>
 #include <QString>
 
+#include "InputCommon/ControllerEmu/Setting/NumericSetting.h"
+
 class MappingWidget;
 
-namespace ControllerEmu
+class MappingDouble : public QDoubleSpinBox
 {
-class NumericSetting;
-}
+public:
+  MappingDouble(MappingWidget* parent, ControllerEmu::NumericSetting<double>* setting);
+
+private:
+  void fixup(QString& input) const override;
+
+  void ConfigChanged();
+
+  ControllerEmu::NumericSetting<double>& m_setting;
+};
 
-class MappingNumeric : public QSpinBox
+class MappingBool : public QCheckBox
```

Comment:
NBD but it feels odd that a boolean is under 'numeric', I'm not sure I would think to look there first.

Reply:
Maybe a little weird. I think bools can be considered numbers. I plan on allowing the checkbox to be transformed into a controller interface expression in the future so maybe it will make sense later? :P